src/dr_llm/pool/store.py

```
@@ -0,0 +1,985 @@
+                raise
+
+    def fail_pending(self, *, pending_id: str, reason: str) -> None:
+        """Mark pending sample as failed."""
+        self.init_schema()
+        tbl = self._schema.pending_table
+        with self._runtime.conn() as conn:
+            conn.execute(
+                _q(
+                    f"UPDATE {tbl} SET status = %s, "
+                    f"metadata_json = jsonb_set(COALESCE(metadata_json, '{{}}'), '{{fail_reason}}', %s::jsonb) "
+                    f"WHERE pending_id = %s"
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)